### PR TITLE
OSDOCS#6630: second iteration of how updates work doc

### PR DESCRIPTION
**Pull request by skopacz1**
[OSDOCS-6630](https://issues.redhat.com/browse/OSDOCS-6630)

Versions: 4.11+

This PR further refines the documentation for how cluster updates work, since there were a lot of feedback items that were deferred when the original documentation was implemented.

QE review:
- [x] QE has approved this change.

Preview: [How cluster updates work](https://64077--docspreview.netlify.app/openshift-enterprise/latest/updating/understanding_updates/how-updates-work)

### CODE REVIEW

**Comment by skopacz1** [comment]
cc. @wking @petr-muller @LalatenduMohanty

File: modules/update-process-workflow.adoc [hunk 1]
```diff
@@ -27,6 +27,8 @@ The job then extracts the manifests and metadata from the release image to a sha
 Certain conditions can prevent updates from proceeding.
 These conditions are either determined by the CVO itself, or reported by individual cluster Operators that detect some details about the cluster that the Operator considers problematic for the update.
 
+// to do: potentially add an example of a precondition to the bullet above.
```
Comment (skopacz1) [inline modules/update-process-workflow.adoc:30]: This is in reference to [this discussion thread](https://github.com/openshift/openshift-docs/pull/60091#discussion_r1212416618).

File: modules/update-manifest-application.adoc [hunk 2]
```diff
@@ -43,12 +43,16 @@ This is especially true for cluster Operators, which might perform their own upd
 While the additional update actions take place, these cluster Operators temporarily set their `Progressing` condition to `True`.
 ====
 
+// to do: potentially reword the note above to clarify that specific resources are being applied at one time, and not necessarily all the resources for that component.
```
Comment (skopacz1) [inline modules/update-manifest-application.adoc:46]: This is in reference to [this discussion thread](https://github.com/openshift/openshift-docs/pull/60091#discussion_r1212371011).

File: modules/update-evaluate-availability.adoc [hunk 3]
```diff
@@ -54,6 +54,8 @@ Supported but not recommended updates:
   Message: Nodes with substantial numbers of containers and CPU contention may not reconcile machine configuration https://bugzilla.redhat.com/show_bug.cgi?id=2111817#c22
 ----
 
+// to do: determine whether the rest of the lines in this module should still be included, since this is pretty in-depth even for this sort of descriptive doc, according to Trevor.
```
Comment (skopacz1) [inline modules/update-evaluate-availability.adoc:57]: This is in reference to [this discussion thread](https://github.com/openshift/openshift-docs/pull/60091#discussion_r1212360392).

Comment (wking) [reply modules/update-evaluate-availability.adoc:57]: I'm not sure if doc-tooling allows this, but a possible compromise between my desire to focus on the abstract description and @petr-muller's [desire][1] to give folks a peek under the hood would be to say that everything in the above `oc adm upgrade` output is sourced from ClusterVersion, and link folks over to [the API docs][2] where they can learn about the structure of `status.availableUpdates` on their own.

[1]: https://github.com/openshift/openshift-docs/pull/60091#discussion_r1213221756
[2]: https://docs.openshift.com/container-platform/4.13/rest_api/config_apis/clusterversion-config-openshift-io-v1.html

File: modules/update-process-workflow.adoc [hunk 4]
```diff
@@ -36,6 +38,8 @@ Cluster Operators are updated in separate stages called Runlevels, and the CVO e
 When the CVO deployment is applied, the current CVO pod terminates, and a CVO pod using the new version starts.
 The new CVO proceeds to apply the remaining manifests.
 
+// to do: potentially replace some instances of "apply" in this doc with something like "reconcile" to imply that a lot of these processes are constantly repeating, rather than happening only once.
```
[the remark this answers is not in the document]
Comment (wking) [reply modules/update-process-workflow.adoc:41]: There aren't to many `apply` at the moment:

```console
$ w3m -dump -cols 200 https://64077--docspreview.netlify.app/openshift-enterprise/latest/updating/understanding_updates/how-updates-work | grep -i apply
      □ Using RBAC to define and apply permissions
      □ Applying autoscaling to a cluster
 2. The CVO begins applying the manifests from the release image. Cluster Operators are updated in separate stages called Runlevels, and the CVO ensures that all Operators in a Runlevel finish
    to apply the remaining manifests.
Some actions can take significant time to finish. The CVO waits for the actions to complete in order to ensure the subsequent Runlevels can proceed safely. The process of applying all manifests is
declaring the new version as reconciled in its status.versions. The CVO waits until the ClusterOperator reports an acceptable status, and then it will start applying manifests at Runlevel 25.
```

The first two there are table-of-contents entries from other sections.  I'd recommend changing:

>  The CVO begins applying the manifests from the release image.

to:

> The CVO begins reconciling the manifests from the release image.

Changing:

> The new CVO proceeds to apply the remaining manifests.

to:

> The new CVO proceeds to reconcile the remaining manifests.

Changing:

> The process of applying all manifests is expected to take 60 to 120 minutes in total...

to:

> Initially reconciling the new release's manifests is expected to take 60 to 120 minutes in total...

Changing:

> The CVO waits until the ClusterOperator reports an acceptable status, and then it will start applying manifests at Runlevel 25.

to:

> The CVO waits until the ClusterOperator reports an acceptable status, and then it will start reconciling manifests at Runlevel 25.

Push-back welcome, "apply" vs. "reconcile" is a nit-level issue for me, and I'm fine leaving it unaddressed.  I'm also fine going through a few rounds of rewording before finding something everyone likes, or before giving up on finding something everyone likes and leaving it as it stands ;)

File: modules/update-manifest-application.adoc [hunk 5]
```diff
@@ -43,12 +43,16 @@ This is especially true for cluster Operators, which might perform their own upd
...
+
 The CVO waits until all cluster Operators in the Runlevel meet the following conditions before it proceeds to the next Runlevel:
 
 * The cluster Operators have an `Available=True` condition.
 
 * The cluster Operators have a `Degraded=False` condition.
 
+// to do: potentially clarify that this condition is not applicable during installations, and also potentially add documentation (here or elsewhere) that explains how the CVO is constantly reconciling states whether or not an update is happening.
```
[the remark this answers is not in the document]
Comment (petr-muller) [reply modules/update-manifest-application.adoc:54]: Maybe I'd really add a "Cluster Version Operator"  section right to the start of "How cluster updates work." page. It would be fairly high-level and it would mention 1) that it is and operator, 2) it applies manifests from release images (also sounds like "Release Images" should be above the "Evaluation of..." one?) 3) how it does it during installation, over the course of normal cluster operation, and during the upgrade - and the upgrade mention sould be accompanied  with a link to the detailed procedure that we already have.

File: modules/update-process-workflow.adoc (hunk 1 above)
Comment (petr-muller) [reply modules/update-process-workflow.adoc:30]: I think coming up with a specific example would be helpful, I'll try to provide one

File: modules/update-process-workflow.adoc (hunk 4 above)
[the remark this answers is not in the document]
Comment (petr-muller) [reply modules/update-process-workflow.adoc:41]: I don't have a strong opinion on this one. I think this could synergize with the "Maybe I'd really add a "Cluster Version Operator" section" comment above? If we explain better that CVO works continuously it can help conveying the idea of "reconciling" as a long-term process of neverending convergence towards the desired state...

File: modules/update-evaluate-availability.adoc (hunk 3 above)
Comment (wking) [reply modules/update-evaluate-availability.adoc:57]: I'm ok with a separate "how does `oc adm upgrade` work?" with the paired examples that demonstrate that currently it's just a pretty-printer for ClusterVersion status.  That is interesting information for folks who think the command is too magical.  I just don't think that delving into that implementation belongs in this *Evaluation of update availability* section.

File: modules/update-process-workflow.adoc (hunk 1 above)
Comment (wking) [reply modules/update-process-workflow.adoc:30]: one possible option would be [`LowDesiredVersion`][1].

[1]: https://github.com/openshift/cluster-version-operator/pull/918/files#diff-1008d196a61fa8941a8df069ea3bd53275cc2f4e0b7d588529485db5598ed02fR58

File: modules/update-evaluate-availability.adoc (hunk 3 above)
Comment (LalatenduMohanty) [reply modules/update-evaluate-availability.adoc:57]: We should discourage folks to run commands like `oc get clusterversion version -o json | jq '.status.availableUpdates'` unless there is no way they can get some information. If the information can retrieved through `oc command` then we do not need to talk about directly querying clusterversion . Because we do not users modifying clusterversion or any underlying resources. If they get in to habit of doing that it is risky for multiple reasons. The UX is hard. They need to parse information which might not be useful for them. They accidentally modify a resource which might cause issues because QE does not test by directly modifying or querying the resources.

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

File: modules/update-evaluate-availability.adoc (hunk 3 above)
Comment (petr-muller) [reply modules/update-evaluate-availability.adoc:57]: > We should discourage folks to run commands like ... unless there is no way they can get some information.

It's hard to make people understand how upgrade works without actually showing them the guts of `ClusterVersion`. Context is important. These examples are presented in the context of _"run command to see internals / how things work"_, not _"run command to get the list of available updates"_.

>  I just don't think that delving into that implementation belongs in this Evaluation of update availability section

I agree with this. My synthesis of all these opinions is that we could have a short section on the `ClusterVersion` object itself: its existence, the fact that you should never modify it directly, that CVO operates over it and that `oc adm upgrade` pretty prints it. Then we could clean the other sections from mentioning the resource.

File: modules/update-cluster-version-object.adoc [hunk 6]
```diff
@@ -0,0 +1,116 @@
+// Module included in the following assemblies:
+//
+// * updating/understanding_updates/how-updates-work.adoc
+
+:_content-type: CONCEPT
+[id="update-cluster-version-object_{context}"]
+= The ClusterVersion object
+
+One of the resources that the Cluster Version Operator (CVO) monitors is `ClusterVersion`.
```
Comment (wking) [inline modules/update-cluster-version-object.adoc:9]: nit: the CVO monitors many resources, but I'd break them into a few classes:

* Resources that the CVO manages, where we are opinionated about some `metadata` and `spec` properties. 
 Some other controller attempts to continuously deliver the CVO's requests and reports the results in `status`.  The CVO consumes `status` and bubbles up any issues in ClusterVersion's `status`.
 * Resources that configure the CVO's operation.  Mostly this is ClusterVersion, where folks who want to manage the CVO set `spec`, and the CVO attempts to continuously deliver those requests and reports the results in `status`.  But the CVO also consumes configuration from other resources, e.g. Proxy, to figure out how the cluster admins want us to reach off-cluster for new update recommendations and similar.
* A small number of resources are both managed by and configuring the CVO, e.g. the `admin-gates` ConfigMap, where we set some keys we care about, but allow the cluster admin to inject additional keys, and we consume and respect those additions too.

Can we start hinting at this with:

> The ClusterVersion resource is the primary interface for managing the Cluster Version Operator (CVO).  Cluster administrators and other controllers may declare their desired state through ClusterVersion `spec` and hear about how the CVO is delivering those requests in `status`.

or something else that says "this is where you steer the CVO" without dropping into "and the CVO in turn will go manage some other resources for you" sorts of things?  We can get at those other resources later, but I think pulling them in now seems overly complicated.

File: modules/update-cluster-version-object.adoc [hunk 7]
```diff
@@ -0,0 +1,116 @@
...
+
+`ClusterVersion` is a custom resource object that contains information relating to the cluster's version, such as the current and desired versions of the cluster.
+When the CVO observes that the desired version does not match the current version in the `ClusterVersion` resource, it attempts to initiate an update to reconcile the cluster with this new desired state.
```
Comment (wking) [inline modules/update-cluster-version-object.adoc:12]: can we de-emphasize updates here with something like:

> The CVO continually reconciles the cluster with the target state declared in ClusterVersion `spec`.  When the desired release differs from the current one, that reconciliation updates the cluster.

to make it clear that "updating or not?" is a subset of the reconciliation the CVO is always doing?

File: modules/update-cluster-version-object.adoc [hunk 8]
```diff
@@ -0,0 +1,116 @@
...
+
+
+//to-do: this might be heading overload, consider deleting this heading if the context switch from the previous paragraph to this content is smooth enough to not require one.
```
Comment (wking) [inline modules/update-cluster-version-object.adoc:15]: one way to  structure would be a generic "consume `spec`, reconcile, report in `status`"  to underline how we match [the usual Kubernetes pattern][1].  Another way to structure would be to have a few sections:

* Reconciling the currently accepted target release, which effectively consumes `status.desired` and reports via `Progressing`, `Failing`, etc.
* Providing next-hop advice, which consumes `spec.upstream` and `channel` and reports in `status.availableUpdates` and `conditionalUpdates` and `RetrievedUpdates`.
  * Also in this line, we're consuming ClusterOperator `Upgradeable` and producing ClusterVersion `Upgradeable`.
* Accepting a proposed next hop, which consumes `spec.desiredUpdate`, `status.availableUpdates`, `conditionalUpdates`, `Upgradeable`, and release signatures, and reports in `status.desired` and `RetrievePayload`.

Each of those touches up against admin activity.  During an update, an admin will bump up against all of those controller loops.  Outside of updates, admins will mostly care if  there are issues reconciling the currently accepted target release, until they start planning and preparing for their next round of updates.

I'm personally agnostic about whether it's easier to explain ClusterVersion as a generic Kube `spec`/`status` resource that happens to be about cluster reconciliation and updates, or if it's easier to explain ClusterVersion as interacting with a series of controllers broken down by use-case.

[1]: https://kubernetes.io/docs/concepts/overview/working-with-objects/#object-spec-and-status

File: modules/update-cvo.adoc [hunk 9]
```diff
@@ -0,0 +1,12 @@
+// Module included in the following assemblies:
+//
+// * updating/understanding_updates/how-updates-work.adoc
+
+:_content-type: CONCEPT
+[id="update-cvo_{context}"]
+= The Cluster Version Operator
+
+// adding a poorly written, technically inaccurate skeleton of a module for now, which can be replaced/refined by SMEs as they see fit
```
Comment (wking) [inline modules/update-cvo.adoc:9]: Your following paragraph looks technically accurate to me.  You  could echo the ClusterVersion sentence about ClusterVersion being the API between the CVO and its managers if you wanted, but you could leave it off too.

Personally, the "CVO is constantly comparing ... and reconciling..." bit is the main thing, and install, update, and normal operations are all minor variations on that theme.  But :shrug:, if we cover that in the second sentence instead of the first, like you have now, that's enough for me.

File: modules/update-evaluate-availability.adoc [hunk 10]
```diff
@@ -19,96 +19,3 @@ If the CVO finds that the cluster does not match the risks of an update, or that
 
 The user interface, either the web console or the OpenShift CLI (`oc`), presents this information in sectioned headings to the administrator.
```
Comment (wking) [inline modules/update-evaluate-availability.adoc:20]: do we want to link out from this concept section to "here's the docs for driving those interfaces to consume this infomation"?  Or do only do procedure -> context links, and not context -> proceedure links?

File: modules/update-cluster-version-object.adoc (hunk 7 above)
Comment (skopacz1) [reply modules/update-cluster-version-object.adoc:12]: Does "the desired release" refer to anything beyond the desired version for the entire cluster? 

Also, if "that reconciliation updates the cluster" can refer to processes outside of the update process described in this "how updates work" assembly, I feel like using the word "update" to mean something other than "cluster-wise change from version X to version X+1" might be confusing to users. But I'm not sure.

File: modules/update-cluster-version-object.adoc (hunk 6 above)
Comment (skopacz1) [reply modules/update-cluster-version-object.adoc:9]: I'll replace the first sentence in the following paragraph with this (and maybe still find a way to mention that it contains version info)

File: modules/update-cvo.adoc (hunk 9 above)
Comment (skopacz1) [reply modules/update-cvo.adoc:9]: I could go either way. If you're fine with how it is, I think it's stronger to start off this "how updates work" page with a sentence about the CVO being the facilitator of updates, and then a clarification that really the CVO does this sort of reconciliation all the time.

But I also understand if that logic sounds backwards and you, or others, want me to emphasize the general reconciliation before diving in to its relevance to updates

File: modules/update-evaluate-availability.adoc (hunk 10 above)
Comment (skopacz1) [reply modules/update-evaluate-availability.adoc:20]: It's tricky, do you mean linking to the [CLI](https://docs.openshift.com/container-platform/4.14/updating/updating_a_cluster/updating-cluster-cli.html#update-upgrading-cli_updating-cluster-cli) and [web console](https://docs.openshift.com/container-platform/4.14/updating/updating_a_cluster/updating-cluster-web-console.html#update-upgrading-web_updating-cluster-web-console) update procedures, where there's a step or two showing how to view the available updates?

I can't link inline in this module, it would have to be in an additional resources list at the end of the section. And if you mean to link to these pages, the context might be lost by the time they get to the additional resources section and see links to update procedures. Maybe I can preface the links with "To learn more about viewing available updates, see the following....".

**Comment by petr-muller** [comment]
LGTM

**Review by LalatenduMohanty — CHANGES_REQUESTED** [review]

File: modules/update-cluster-version-object.adoc [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+// * updating/understanding_updates/how-updates-work.adoc
+
+:_content-type: CONCEPT
+[id="update-cluster-version-object_{context}"]
+= The ClusterVersion object
+
+One of the resources that the Cluster Version Operator (CVO) monitors is `ClusterVersion`.
+
+The `ClusterVersion` custom resource object is the primary interface for managing the CVO.
+Cluster administrators and other controllers can declare their desired state through `ClusterVersion` `spec` and observe how the CVO is delivering those requests in `status`.
```
[the remark this answers is not in the document]
Comment (LalatenduMohanty) [reply modules/update-cluster-version-object.adoc:12]: My main concern is with the text that says "cluster administrators can declare their desired state through `clusterversion`. @wking WDYT?

[the remark this answers is not in the document]
Comment (wking) [reply modules/update-cluster-version-object.adoc:12]: This sentence is just pointing out that ClusterVersion follows [Kubernetes' usual `spec`/`status` pattern][1].  I think it's worth pointing out that the intended data flows are:

* Admin desires -> ClusterVersion `spec` declarations -> CVO attempts to deliver the desired state.
* CVO has opinions on current state and progress -> ClusterVersion `status` reporting -> admin clarity on current situation.

but I'm open to rephrasing if we can express those two directions of data flow more clearly.

[1]: https://kubernetes.io/docs/concepts/overview/working-with-objects/#object-spec-and-status

[the remark this answers is not in the document]
Comment (LalatenduMohanty) [reply modules/update-cluster-version-object.adoc:12]: Here is my suggested text. 

OpenShift components and administrators can communicate/interact with CVO through `ClusterVersion` object. The desired CVO state should be declared through the `Clusterversion` object and the current CVO state can be seen through status of the `ClusterVersion` object. 

Note: We do not suggest users to directly modify the `ClusterVersion` object. They should use the standard interfaces e.g. CLI and web console to declare their desired update etc.

**Comment by skopacz1** [comment]
@shellyyang1989 could you PTAL when you have a chance? Thanks!

**Comment by shellyyang1989** [comment]
LGTM

File: modules/update-cluster-version-object.adoc [hunk 12]
```diff
@@ -0,0 +1,123 @@
+// Module included in the following assemblies:
+//
+// * updating/understanding_updates/how-updates-work.adoc
+
+:_content-type: CONCEPT
+[id="update-cluster-version-object_{context}"]
+= The ClusterVersion object
+
+One of the resources that the Cluster Version Operator (CVO) monitors is `ClusterVersion`.
```
Comment (kelbrown20) [inline modules/update-cluster-version-object.adoc:9]:
```suggestion
One of the resources that the Cluster Version Operator (CVO) monitors is the `ClusterVersion` resource.
```

File: modules/update-cluster-version-object.adoc [hunk 13]
```diff
@@ -0,0 +1,123 @@
...
+
+Administrators and OpenShift components can communicate or interact with the CVO through the `ClusterVersion` object.
+The desired CVO state is declared through the `ClusterVersion` object and the current CVO state is reflected in the object's status.
+
+[NOTE]
+====
+Do not directly modify the `ClusterVersion` object. Instead, use interfaces such as the `oc` CLI or the web console to declare your desired update target.
```
Comment (kelbrown20) [inline modules/update-cluster-version-object.adoc:16]: [IBM style suggests](https://www.ibm.com/docs/en/ibm-style?topic=word-usage#d) not to use `desire` 
```suggestion
Do not directly modify the `ClusterVersion` object. Instead, use interfaces such as the `oc` CLI or the web console to declare your update target.
```

File: modules/update-cluster-version-object.adoc [hunk 14]
```diff
@@ -0,0 +1,123 @@
...
+====
+
+The CVO continually reconciles the cluster with the target state declared in `ClusterVersion` `spec`.
```
Comment (kelbrown20) [inline modules/update-cluster-version-object.adoc:19]: Not sure if its suppose to be `ClusterVersion.spec`.

File: modules/update-cluster-version-object.adoc [hunk 15]
```diff
@@ -0,0 +1,123 @@
...
+When the desired release differs from the current one, that reconciliation updates the cluster.
```
Comment (kelbrown20) [inline modules/update-cluster-version-object.adoc:20]: desired -> required

File: modules/update-cluster-version-object.adoc [hunk 16]
```diff
@@ -0,0 +1,123 @@
...
+
+//to-do: this might be heading overload, consider deleting this heading if the context switch from the previous paragraph to this content is smooth enough to not require one.
+[discrete]
+== Update availability data
+
+The `ClusterVersion` resource also contains information about updates that are available to the cluster.
+This includes updates that available but not recommended due to a known risk that applies to the cluster, which are known as conditional updates.
```
Comment (kelbrown20) [inline modules/update-cluster-version-object.adoc:27]:
```suggestion
This includes updates that available, but not recommended, due to a known risk that applies to the cluster that are known as conditional updates.
```

File: modules/update-cluster-version-object.adoc [hunk 17]
```diff
@@ -0,0 +1,123 @@
...
+To learn about how the CVO maintains this information about available updates in the `ClusterVersion` resource, see the "Evaluation of update availability" section.
```
Comment (kelbrown20) [inline modules/update-cluster-version-object.adoc:28]:
```suggestion
To learn how the CVO maintains this information about available updates in the `ClusterVersion` resource, see the "Evaluation of update availability" section.
```

File: modules/update-cluster-version-object.adoc [hunk 18]
```diff
@@ -0,0 +1,123 @@
...
+
+You can inspect all available updates with the following command:
+
+[source,terminal]
+----
+$ oc adm upgrade --include-not-recommended
+----
+
+The additional `--include-not-recommended` parameter includes updates that are available but not recommended due to a known risk that applies to the cluster.
```
Comment (kelbrown20) [inline modules/update-cluster-version-object.adoc:37]: Wondering if this would work better as a IMPORTANT block, but WDYT?

Edit: Might not be needed as a important block if you reformat the commands and outputs

File: modules/update-cluster-version-object.adoc [hunk 19]
```diff
@@ -0,0 +1,123 @@
...
+
+//to-do: this might be heading overload, consider deleting this heading if the context switch from the previous paragraph to this content is smooth enough to not require one.
+[discrete]
+== Update availability data
+
+The `ClusterVersion` resource also contains information about updates that are available to the cluster.
+This includes updates that available but not recommended due to a known risk that applies to the cluster, which are known as conditional updates.
+To learn about how the CVO maintains this information about available updates in the `ClusterVersion` resource, see the "Evaluation of update availability" section.
+
+You can inspect all available updates with the following command:
```
Comment (kelbrown20) [inline modules/update-cluster-version-object.adoc:30]: I think these commands and outputs can look more organized if you kinda separate them. My suggestion would be to have this statement as a bullet point (add the `+` as needed) then have another bullet point at line 66 with this statement
"One way to directly inspect the underlying availability data created by the CVO is by querying the ClusterVersion resource with the following command:"
(Also add `+` as needed). 

Some of the statements I think get lost in the blocks and they need to be more sectioned

File: modules/update-cvo.adoc [hunk 20]
```diff
@@ -0,0 +1,12 @@
...
+
+The Cluster Version Operator (CVO) is the primary component that orchestrates and facilitates the {product-title} update process.
+During installation and normal cluster operation, the CVO is constantly comparing the manifests of managed cluster Operators to in-cluster resources, and reconciling discrepancies to ensure that the actual state of these resources match their desired state.
```
Comment (kelbrown20) [inline modules/update-cvo.adoc:12]:
```suggestion
During installation and standard cluster operation, the CVO is constantly comparing the manifests of managed cluster Operators to in-cluster resources, and reconciling discrepancies to ensure that the actual state of these resources match their desired state.
```

**Review by kelbrown20 — COMMENTED** [review]
Hey Seb! Just a couple wording changes and suggestions for formating. Great work on this!

File: modules/update-cluster-version-object.adoc (hunk 14 above)
Comment (skopacz1) [reply modules/update-cluster-version-object.adoc:19]: So these are intentionally spaced out as separate words, but I can try and reword this to have more clarity. Maybe "the `spec` property of the `ClusterVersion` resource"

File: modules/update-cluster-version-object.adoc (hunk 15 above)
Comment (skopacz1) [reply modules/update-cluster-version-object.adoc:20]: "desired" in this instance doesn't refer to user intention, but rather the desired state of the cluster, where the reconciliation process tries to make the actual state match the desired state. So when the desired version is different from the actual version, the reconciliation to match the desired state is the actual cluster update process we're documenting here.

I can reword to make this more clear, but I think keeping "desired" in here is important for describing how the process works, despite the ISG guidance. Let me know what you think!

File: modules/update-cluster-version-object.adoc (hunk 14 above)
Comment (kelbrown20) [reply modules/update-cluster-version-object.adoc:19]: Gotcha, I really wasn't too sure on this one but I had a thought. But that sounds good to me 👍

File: modules/update-cluster-version-object.adoc (hunk 15 above)
Comment (kelbrown20) [reply modules/update-cluster-version-object.adoc:20]: Ahh I get what you're saying, yeah I think that the reword would be great!

**Review by jldohmann — APPROVED** [review]
LGTM

**Comment by jldohmann** [comment]
@skopacz1 it looks like all the auto CPs to every branch but 4.14 failed, so you'll need to manually CP. Please lmk if you have any questions and feel free to ping me once those CPs are up and i'll merge them 🙂 thanks!
